src/plugins/analysis/kernel_config/code/kernel_config.py

```diff
@@ -100,3 +114,51 @@ def object_is_kernel_image(file_object: FileObject) -> bool:
         return 'software_components' in file_object.processed_analysis and \
                'summary' in file_object.processed_analysis['software_components'] and \
                any('linux kernel' in component.lower() for component in file_object.processed_analysis['software_components']['summary'])
+
+    @staticmethod
+    def check_kernel_config(kernel_config: str) -> dict:
+        try:
+            with NamedTemporaryFile() as fp:
+                fp.write(kernel_config.encode())
+                fp.seek(0)
+                command = f'{CHECKSEC_PATH} --kernel={fp.name} --output=json 2>/dev/null'
+                output = execute_shell_command(command)
+                result = json.loads(output)
+                whitelist_configs(result)
+                return result
+
+        except (JSONDecodeError, KeyError):
+            logging.debug('Checksec kernel analysis failed')
+        return {}
+
+
+def whitelist_configs(config_results: dict) -> dict:
+    kernel_whitelist = ['kernel_heap_randomization', 'gcc_stack_protector', 'gcc_stack_protector_strong',
```

Comment:
I'd move this up into a global variable (i.e. `KERNEL_WHITELIST`)

Reply:
changed